cmd/server/server.go

```diff
@@ -77,40 +73,38 @@ func Run(ctx context.Context) (err error) {
 
 	router.PathPrefix("/debug/").Handler(http.DefaultServeMux)
 
+	//
 	var srv = http.Server{Addr: Host, Handler: router}
 
-	idleConnsClosed := make(chan struct{})
-	// канал для перенаправления прерываний
-	// поскольку нужно отловить всего одно прерывание,
-	// ёмкости 1 для канала будет достаточно
-	sigint := make(chan os.Signal, 1)
-	// регистрируем перенаправление прерываний
-	signal.Notify(sigint, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+	ctx, cancel := context.WithCancel(context.Background())
 
-	// запускаем горутину обработки пойманных прерываний
 	go func() {
-		// читаем из канала прерываний
-		// поскольку нужно прочитать только одно прерывание,
-		// можно обойтись без цикла
-		<-sigint
-		// получили сигнал os.Interrupt, запускаем процедуру graceful shutdown
-		if err := srv.Shutdown(ctx); err != nil {
-			// ошибки закрытия Listener
-			log.Printf("HTTP server Shutdown: %v", err)
-		}
-		// сообщаем основному потоку,
-		// что все сетевые соединения обработаны и закрыты
-		close(idleConnsClosed)
+		exit := make(chan os.Signal, 1)
+		signal.Notify(exit, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+		<-exit
+		cancel()
+	}()
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() error {
+		return srv.ListenAndServe()
```

Comment:
горутина ничего не может никуда вернуть. Правильнее все-таки
 ```suggestion
		if err := srv.ListenAndServe(); err != nil && анализ типа ошибки {
			log.Fatal(err)
		}
```
Надо повнимательней посмотреть, что возвращает сервер в случае вызова его шатдауна. Возможно, никакого анализа не надо. Но, если, например, порт занят, сервер должен упасть!